File: 169 - majorityElement.js
const nums = [3, 3, 4];

var majorityElement = function (nums) {
  let count = 0;
  let element;

  const holder = {};

  for (let i = 0; i < nums.length; i++) {
    if (holder[nums[i]] === undefined) {
      holder[nums[i]] = 1;
    } else {
      holder[nums[i]] = holder[nums[i]] + 1;
    }
    if (holder[nums[i]] > count) {
      console.log(holder[nums[i]]);
      count = holder[nums[i]];
      element = nums[i];
    }
    console.log(count, element, nums[i]);
  }
  return element;
};

console.log(majorityElement(nums));

File: 169 - majorityElement.py
input = [2,2,1,1,1,2,2];

class Solution:
    def majorityElement(self, nums):
        holder = {}
        for number in nums:
            if number not in holder:
                holder[number] = 0
            else:
                holder[number] += 1
            # alternate solution:
            # have a count and element variable that we update
            # with the highest count and element with highest count
            # as we iterate through the list
            # if holder[number] >= count:
            #     count += 1
            #     element = number

        return max(holder, key=holder.get)

solution = Solution()
print(solution.majorityElement(input))